### PR TITLE
refactor(manager): use "count criteria" to get total count in search

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

File: graviti/manager/commit.py [hunk 1]
```diff
@@ -162,7 +163,17 @@ def _getter(offset: int, limit: int) -> Dict[str, Any]:
                 limit=limit,
             )
 
-        total_count = _getter(0, 1)["total_count"]
+        count_criteria = deepcopy(criteria)
+        count_criteria["select"] = [{"count": {"$count": ["$.*"]}}]
```
Comment (graczhual) [inline graviti/manager/commit.py:167]: do we need to set limit = 1 here?

File: graviti/manager/commit.py [hunk 2]
```diff
@@ -162,7 +163,17 @@ def _getter(offset: int, limit: int) -> Dict[str, Any]:
                 limit=limit,
             )
 
-        total_count = _getter(0, 1)["total_count"]
+        count_criteria = deepcopy(criteria)
```
Comment (linjiX) [inline graviti/manager/commit.py:166]: why we need `deepcopy` here?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/commit.py (hunk 1 above)
Comment (AChenQ) [reply graviti/manager/commit.py:167]: Dont't need limit here, just the count will be returned.

File: graviti/manager/commit.py (hunk 2 above)
Comment (AChenQ) [reply graviti/manager/commit.py:166]: To prevent the `criteria` from changing.

**Review by graczhual — APPROVED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**